modules-sdks/pax-terminal/instore-solution/java.md

```diff
@@ -0,0 +1,311 @@
+      logOutput("Logout successful");
+    } else {
+      logOutput("Logout failed with the following error: " + r.getErrorCondition());
+    }
+  });
+}
+```
+
+### Abort
+
+Call *abort* to abort a payment transaction.
+```java
+public void onAbortButtonClick() {
+  logOutput("Aborting payment...");
+
+  CompletableFuture<AbortRequestResult> payment = paxTerminal.abortAsync();
+  payment.thenAccept(this::showAbortResultMessage);
+}
+
+private void showAbortResultMessage(AbortRequestResult result) {
+  if (result.isSuccess()) {
+    logOutput("Abort ok");
+  } else {
+    logOutput("Abort was not ok. Try to reverse payment.");
+  }
+}
+```
+
+## Callbacks
+
+To receive event callbacks from the terminal the following methods need to be implemented
+in the interface `SwpTrmCallbackInterface`.
+
+### Terminal Display Event
+
+Get status information and displays from the terminal.
+
+```java
+@Override
+public void terminalDisplayEvent(int txtId, String text) {
+  LOG.info("Terminal event: id {}, text: {}", txtId, text);
+}
+```
+
+### Terminal Notification Event
+
+The event function is used by Swedbank Pay Payment Application to communicate an ‘out of sequence’ event - the beginning of maintenance for example.
+
+```java
+@Override
+public void terminalNotificationEvent(EventToNotifyEnumeration type, String eventDetails) {
+    LOG.info("Terminal notification event: type {}, event details {}", type, eventDetails);
+}
+```
+
+### Terminal Address Obtained Event
+
+Called when a new terminal is installed.
+
+```java
+@Override
+public void terminalAddressObtainedEvent(String ipv4, int port) {
+  LOG.info("Terminal address obtained event: ipv4 {}, port {}", ipv4, port);
+}
+```
+
+### Print Request
+
+Called when a receipt is to be printed. The print result needs to be confirmed from the ECR.
+
+```java
+@Override
+public boolean onPrintRequest(boolean transactionEnded, Receipt receipt) {
+    LOG.info("Print receipt: " + receipt.getCustomerFormattedReceipt());
+
+    if (transactionEnded) {
+      logOutput("Transaction ended");
+    }
+    return true;
+}
+```
+
+### Request For Confirmation Event
+
+Called when there is a confirmation needed by the cashier, i.e when a signature purchase needs
+to be confirmed. Will soon be deprecated when signature purchase is removed.
+
+```java
+@Override
+public void requestForConfirmationEvent(String message, ConfirmationResult cashierConfirmation) {
+    LOG.info("Message: " + message)
+    //Confirm with true if the signature matches
+    cashierConfirmation.confirm(true);
+}
+```
+
+## Logging
+
+Logging is done through Slf4j. To see what is going on set DEBUG level for the following package `com.swedbankpay`.
+
+When running the application standalone it can be handy to se the communication between the ECR and the terminal.
+To log communication between the ECR and terminal to some text window in the application the following can be used. This can be enabled or disabled runtime. It is disabled by default.
```

Comment:
Wrap lines at 80 characters, please. :)

```suggestion
To log communication between the ECR and terminal to some text window in the
application the following can be used. This can be enabled or disabled runtime.
It is disabled by default.
```